test/constructorio-search.js

```diff
@@ -0,0 +1,417 @@
+      const constructorio = new Constructorio(testConfig);
+      const query = 'xylophone';
+
+      constructorio.getSearchResults({
+        query,
+        section: 'Products',
+        ...personalizationParameters,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response.response).to.have.property('results').that.is.an('array').length(0);
+        done();
+      });
+    });
+
+    it('should return error when retrieving search results with invalid section', (done) => {
+      const constructorio = new Constructorio(testConfig);
+      const query = 'xylophone';
+
+      constructorio.getSearchResults({
+        query,
+        section: 'invalid',
+        ...personalizationParameters,
+      }, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message', 'Unknown section: invalid');
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return one search result when specifying a num results per page parameter', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getSearchResults({
+        query: 'drill',
+        section: 'Products',
+        num_results_per_page: 1,
+        ...personalizationParameters,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response.response).to.have.property('results').that.is.an('array').length(1);
+        done();
+      });
+    });
+
+    it('should return error when retrieving search results with invalid num results per page parameter', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getSearchResults({
+        query: 'drill',
+        section: 'Products',
+        num_results_per_page: 'abc',
+        ...personalizationParameters,
+      }, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message', 'num_results_per_page must be an integer');
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return search results when specifying a valid page parameter', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getSearchResults({
+        query: 'drill',
+        section: 'Products',
+        page: 1,
+        ...personalizationParameters,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response.response).to.have.property('results').that.is.an('array').length.to.be.above(0);
+        done();
+      });
+    });
+
+    it('should return error when retrieving a listing of groups with invalid page parameter', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getSynonymGroups({
```

Comment:
did you mean to call `getSearchResults` here instead of `getSynonymGroups`?

probably need to fix wording in test description as well

Reply:
Yes I did!